src/smsCommandWithDataSet/CommandEditWithDataSetForm.js

```diff
@@ -164,6 +164,135 @@ const formatSmsCodes = updates => {
     }
 }
 
+const FormComponent = ({
+    DE_COC_combination_data,
+    dataTest,
+    handleSubmit,
+    onCancel,
+    pristine,
+    selectedDataSetOption,
+}) => {
+    const form = useForm()
+    const hasSpecialChars = useMemo(() => {
+        const specialChars = form.getState().values[
+            FIELD_COMMAND_SPECIAL_CHARS_NAME
+        ]
+
+        return specialChars.length
+    }, [])
```

Comment:
Is it necessary to memoize this? Did you notice things getting slow when you didn't memoize?

Reply:
Fair point, I haven't noticed any performance issues, this is legacy code from a previous, alternative approach. I'll remove it

Reply:
Reason I'm mentioning it is because of the reasons outlined here: https://kentcdodds.com/blog/usememo-and-usecallback. It doesn't always help so I tend to favor leaving it out until its both needed and benchmarked.